contracts/Forwarder.sol

```diff
@@ -62,6 +62,21 @@ contract Forwarder {
     flush();
   }
 
+  /**
+   * ERC721 standard callback function for when a ERC721 is transfered. The forwarder will send the nft
+   * to the base wallet once the nft contract invokes this method after transfering the nft.
+   *
+   * @param _operator The address of the nft contract
+   * @param _from The address of the sender
+   * @param _tokenId The token id of the nft
+   * @param _data Additional data with no specified format, sent in call to `_to`
+   */
+  function onERC721Received(address operator, address from, uint256 tokenId, bytes memory data) public virtual override returns (bytes4) {
+    IERC721(_operator).safeTransferFrom(from, parentAddress, tokenId, data);
+
```

Comment:
IMO this is quite dangerous -- since it's public, it basically allows anyone to call `safeTransferFrom` on any contract with arbitrary params (including data but excluding recipient address)

Reply:
true, so would something like `require(msg.sender == operator)` mitigate that risk?

Reply:
so this should only allow us to `safeTransferFrom` the receive address `address(this)` to the parent wallet `parentAddress` 

I think toda's right though, maybe we put a ERC165 check first to make sure the callee supports `0x80ac58cd`

https://eips.ethereum.org/EIPS/eip-721 if you look at that spec that strongly identifies a 721

Reply:
maybe a re-entrancy guard too?